### PR TITLE
adds option to skip over rows before setting headers [useful when there are rows of junk data at the top of CSV]

### PR DESCRIPTION
**Pull request by jekrb**
Greetings :wave:

example:

```
// junk_rows.csv
,somejunk,<! />
,nope,
yes,yup,yeah
ok,ok,ok!
```

I want the `yes,yup,yeah` row to be the header, and for the rows above it to be ignored.
With this PR we can do something like this:

```
fs.createReadStream('junk_rows.csv')
  .pipe(csv({ skipUntil: 'yup' }) // header becomes yes,yup,yeah
  .pipe(process.stdout) // {yes: 'ok', yup: 'ok', yeah: 'ok!'}
```

I'm making this PR because I had an issue with some CSV files containing junk data at the top, and I wanted to programmatically skip over those rows rather than manually deleting them.

I saw the mapHeaders option, but that didn't look entirely useful for me because the CSVs I have are so wide that I wasn't entirely sure what all the headers were.  So I needed something that basically said "look at each cell in the row, if one of the cells contain a certain string which should be a header, make that row the header."

Thanks for making this module! 🌴  🌴

### CODE REVIEW

**Comment by shellscape** [comment]
@jekrb it's been a while since you put this PR together. Do you have interest in updating it?

**Comment by jekrb** [comment]
@shellscape sure thing! I just updated to fix the conflicts and test.

**Comment by jekrb** [comment]
@shellscape I didn't think of skipping line numbers before, but now that you mention it, I do like the idea! I think that would be much clearer than looking for a value in a cell of a row.

**Comment by shellscape** [comment]
Cool. Would you like to update this PR to skip line numbers instead?

**Comment by jekrb** [comment]
Yeah that sounds like a good idea to me 👍

File: index.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ var Parser = function (opts) {
   this.strict = opts.strict || null
   this.mapHeaders = opts.mapHeaders || identity
   this.mapValues = opts.mapValues || identity
+  this.skipUntil = opts.skipUntil || null
```
Comment (shellscape) [inline index.js]: let's use `skipLines` here instead, as that's more descriptive of the feature

Comment (jekrb) [reply index.js]: all set!

**Comment by jekrb** [comment]
CLI flag makes sense to me, I'll add that too

**Comment by jekrb** [comment]
@shellscape I've added CLI flags and updated the readme 😄

File: test/test.js [hunk 2]
```diff
@@ -345,6 +345,15 @@ test('format values', (t) => {
   }
 })
 
+test('skip rows until', function (t) {
+  collect('junk_rows.csv', {skipLines: 3}, verify)
```
Comment (shellscape) [inline test/test.js]: If `skipLines` is `3` I would expect that the headers line would then be `ok,ok,ok!`. It looks like you need to account for the zero-based index there. To get the result the test is expecting, I would expect to pass `2` to the option.

**Comment by jekrb** [comment]
@shellscape I updated the indexing and migrated my test to ava 👍

**Pull request merged**